src/Stratis.Bitcoin/Consensus/Rules/ConsensusRule.cs

```diff
@@ -24,25 +21,32 @@ public abstract class ConsensusRule : IConsensusRule
         public virtual void Initialize()
         {
         }
+    }
 
+    /// <summary>An abstract rule for implementing consensus rules.</summary>
+    public abstract class SyncConsensusRule : ConsensusRuleBase, ISyncBaseConsensusRule
+    {
         /// <summary>
-        /// Execute the logic in the current rule in an async approach.
+        /// Execute the logic in the current rule.
         /// If the validation of the rule fails a <see cref="ConsensusErrorException"/> will be thrown.
         /// </summary>
         /// <param name="context">The context that has all info that needs to be validated.</param>
-        /// <returns>The execution task.</returns>
-        public virtual Task RunAsync(RuleContext context)
+        public virtual void Run(RuleContext context)
```

Comment:
Why not abstract

Reply:
Answered on slack. Copypasted here: 
we need typecasts because interface is empty
But we need to call run or run async
So we are doing this ugly thing